doc/stdenv/meta.chapter.md

```diff
@@ -98,13 +98,32 @@ Hydra will never create prebuilt binaries for these platform types, even if they
 In general it is preferable to set `meta.platforms = lib.platforms.all` and then exclude any platforms on which the package is known not to build.
 For example, a package which requires dynamic linking and cannot be linked statically could use this:
 
+```nix
+{
+  meta.platforms = lib.platforms.all;
+  meta.badPlatforms = [{ hasSharedLibraries = false; }];
+}
+```
+
+or
+
 ```nix
 {
   meta.platforms = lib.platforms.all;
   meta.badPlatforms = [ lib.systems.inspect.platformPatterns.isStatic ];
 }
 ```
 
+In the examples above, `lib.systems.inspect.platformPatterns.isStatic` is an alias for `{ isStatic = true; }` platform pattern. The difference between `isStatic` and `hasSharedLibraries` is that:
+
+* isStatic: packages should be statically linked against the dependencies when
+  possible.
+
+* hasSharedLibraries: the platform supports dynamic linking. For example, if set
+  to `false`, things like dlopen do not work. If the package *requires* dynamic
+  linking (e.g. loads plugins), it should be considered unavailable on such
```

Comment:
```suggestion
  linking (e.g. must load plugins dynamically), it should be considered unavailable on such
```

I don't want to spend too many words on this, but we can hint a bit:

must: plugins may be optional.
dynamically: maybe the plugins can be linked statically; always plugged in. Linux allows this for its modules.